src/linux/frida-helper-glue.c

```diff
@@ -1540,6 +1813,89 @@ frida_resolve_linker_function (pid_t pid, gpointer func)
   return remote_address;
 }
 
+#elif defined (HAVE_UCLIBC)
+
+static GumAddress
+frida_resolve_linker_function (pid_t pid, gpointer func)
+{
+  const gchar * linker_file_name = "libdl";
+  gchar * linker_path;
+  GumAddress local_base, remote_base, remote_address;
+
+  local_base = frida_find_library_base (getpid (), linker_file_name, &linker_path);
+  g_assert (local_base != 0);
+
+  remote_base = frida_find_library_base (pid, linker_file_name, NULL);
+  if (remote_base == 0)
+  {
+    gpointer rpnt, rpnt_next, tpnt;
+    gint32 ret;
```

Comment:
Could we split this `ret` variable into two:

```
gboolean success;
gint res;
```

Then when we g_assert on `success` we'd `g_assert (success)`.